indico/web/forms/fields.py

```diff
@@ -132,9 +141,12 @@ def __init__(self, *args, **kwargs):
         super(IndicoLocationField, self).__init__(*args, **kwargs)
 
     def process_formdata(self, valuelist):
-        super(IndicoLocationField, self).process_formdata(valuelist)
-        self.data['room'] = Room.get(int(self.data['room_id'])) if self.data.get('room_id') else None
-        self.data['venue'] = Location.get(int(self.data['venue_id'])) if self.data.get('venue_id') else None
+        if _preprocessed_formdata(valuelist):
```

Comment:
Not sure if this makes much sense here... LocationField is pretty special and in any case I don't think we'd pass the actual DB objects here.  So at least the Room.get/Location.get lines should always run IMO even if there's preprocessed json data (i.e. just skip the json parsing)